numpy/_array_api/_types.py

```diff
@@ -0,0 +1,23 @@
+"""
+This file defines the types for type annotations.
+
+These names aren't part of the module namespace, but they are used in the
+annotations in the function signatures. The functions in the module are only
+valid for inputs that match the given type annotations.
+"""
+
+__all__ = ['Literal', 'Optional', 'Tuple', 'Union', 'array', 'device',
+           'dtype', 'SupportsDLPack', 'SupportsBufferProtocol', 'PyCapsule']
+
+from typing import Literal, Optional, Tuple, Union, TypeVar
```

Comment:
Is there a reason for re-exporting these `typing` constructs?

Reply:
Good point, shouldn't be needed.

Reply:
I just did it this way so that all the types come from a single import. This made generating the template code a little easier, but if we feel this isn't clean for the final code I can remove it.